subsys/bluetooth/audio/bap_scan_delegator.c

```diff
@@ -494,13 +494,43 @@ static int scan_delegator_add_source(struct bt_conn *conn,
 	uint32_t aggregated_bis_syncs = 0;
 	uint32_t broadcast_id;
 	bool bis_sync_requested;
+	uint16_t total_len;
+	struct bt_bap_bass_cp_add_src *add_src;
 
 	/* subtract 1 as the opcode has already been pulled */
 	if (buf->len < sizeof(struct bt_bap_bass_cp_add_src) - 1) {
 		LOG_DBG("Invalid length %u", buf->size);
 		return BT_GATT_ERR(BT_ATT_ERR_WRITE_REQ_REJECTED);
 	}
 
+	add_src = (void *)buf->data;
+	total_len = sizeof(struct bt_bap_bass_cp_add_src) - 1;
+	for (int i = 0; i < add_src->num_subgroups; i++) {
```

Comment:
Might be worth also checking if `add_src->num_subgroups > BASE_SUBGROUP_MAX_COUNT`, and reject early if it's not a valid number of subgroups. `BASE_SUBGROUP_MAX_COUNT` would need to be moved to a header file though.

Reply:
For the case `add_src->num_subgroups > BASE_SUBGROUP_MAX_COUNT`, it is not an error due to invalid data. It is an error due to insufficient host resources. If this error is checked first, will it mask the error that the received data is invalid? For example, if the `source id` is invalid and `add_src->num_subgroups > CONFIG_BT_BAP_BASS_MAX_SUBGROUPS` in a received packet, the error code should be `BT_BAP_BASS_ERR_INVALID_SRC_ID` instead of `BT_ATT_ERR_INSUFFICIENT_RESOURCES`. If the condition `add_src->num_subgroups > CONFIG_BT_BAP_BASS_MAX_SUBGROUPS` is check first, the returned error is `BT_ATT_ERR_INSUFFICIENT_RESOURCES` instead of `BT_BAP_BASS_ERR_INVALID_SRC_ID`.